src/classes/GiftBatch.cls

```diff
@@ -138,4 +139,25 @@ public inherited sharing class GiftBatch {
     public void updateRecords() {
         gifts.updateRecords();
     }
+
+    public Money total() {
+        //todo: use giftBatch Totals (and place in Selector layer)
+        return new Money(dataImportModel().totalRowAmount, 'US');
+    }
+
+    public Integer totalCount() {
+        //todo: use giftBatch Totals (and place in Selector layer)
+        return dataImportModel().totalCountOfRows;
+    }
+
+    private BGE_DataImportBatchEntry_CTRL.DataImportModel typedModel;
+    private BGE_DataImportBatchEntry_CTRL.DataImportModel dataImportModel() {
```

Comment:
The refactoring work will eventually remove this dependency on the old BGE controller.